### PR TITLE
10.0 migration/account product fiscal classification

### PR DESCRIPTION
**Pull request by renatonlima**
Replaces #62

### CODE REVIEW

**Review by rvalyi — APPROVED** [review]
kudos for the work!

**Comment by rvalyi** [comment]
@renatonlima you made a great part of this module: the initial concept before the refactor from Sylvain, you helped the refactor and also coded all the updates regarding the templates system so I think you deserve to be added back into the the contributors section of the README.

File: account_product_fiscal_classification/models/account_product_fiscal_classification_model.py [hunk 1]
```diff
@@ -33,13 +15,15 @@ class AccountProductFiscalClassificationModel(models.AbstractModel):
...
     code = fields.Char()
-
     name = fields.Char(
-        size=_MAX_LENGTH_NAME, required=True, select=True, translate=True)
-
+        size=_MAX_LENGTH_NAME,
+        required=True,
+        index=True,
+        translate=True
+    )
```
Comment (legalsylvain) [inline account_product_fiscal_classification/models/account_product_fiscal_classification_model.py]: don't understand such changes ? is it an OCA convention to set an argument by line ?

(and other in other python file).

(not a blocking point, just a remark)

File: account_product_fiscal_classification/views/view_product_category.xml [hunk 2]
```diff
@@ -7,41 +7,41 @@
...
             </field>
-        </field>
-    </record>
+          </group>
+        </group>
+      </xpath>
+    </field>
+  </record>
+
+  <record id="view_product_category_tree" model="ir.ui.view">
```
Comment (legalsylvain) [inline account_product_fiscal_classification/views/view_product_category.xml]: Nice improvement, could you add a group to this view. (like accounting manager). I'm not sure "basic" users are interested by this information.

**Review by legalsylvain — APPROVED** [review]
> @renatonlima you made a great part of this module: the initial concept before the refactor from Sylvain, you helped the refactor and also coded all the updates regarding the templates system so I think you deserve to be added back into the the contributors section of the README.

:+1: 

some minor remarks. LGTM otherwise. Thanks !

File: account_product_fiscal_classification/views/view_product_category.xml (hunk 2 above)
Comment (bealdav) [reply account_product_fiscal_classification/views/view_product_category.xml]: Done

Comment (bealdav) [reply account_product_fiscal_classification/views/view_product_category.xml]: classif

**Review by hparfr — APPROVED** [review]

**Review by legalsylvain — CHANGES_REQUESTED** [review]
could you revert bbfc84a598a50aa866006b9d28ede5e864ce67e6 (except the minor changes included in it) ?
It changes tab lenght and it make review harder to do for nothing.
(except if two spaces are a new OCA rules, but I don't think so).

**Review by mourad-ehm — APPROVED** [review]
Code review. Not tested

**Comment by bealdav** [comment]
@legalsylvain You approved 20 Dec 2017 with minor remarks that I fixed. 

I don't understand why revert an xml file. I've made tests on this module, no problem.

I think we can merge as it ? Thanks a lot for this review and to allow going ahead. ;-)

**Comment by legalsylvain** [comment]
Well, the current PR makes a lot of useless changes regarding tabulation. The current OCA norm is using 4 spaces to tab in python or xml views. (that was the state of the module in V8).
with this PR, all tabs are changing to 2 spaces in XML views.
That's why I asked to revert the commit, because this one include all the changes regarding the tab.

Not a big deal but 
1) I don't understand why changing.
2) I don't understand why not using OCA default usage.
3) I see it as a polution during the review process. (see here for exemple : https://github.com/OCA/account-fiscal-rule/pull/74/files#diff-9fcd0a597e15e54c1681c0e5d9015ded)

kind regards.

Nota : I don't want to block anything ! I can do the revert in the akretion repository if you have no time.

**Comment by bealdav** [comment]
Ok thanks for clarifications.

Please @renatonlima could you revert your commit and fix this PR.

Thanks a lot

**Comment by bealdav** [comment]
Rio ne répond plus. pas grave, on va quand même avancer.
@legalsylvain je ne sais pas si cela correspond à ce que tu demandait ?

**Comment by legalsylvain** [comment]
Thanks @bealdav 

It remains a file with the same problem (*), but your PR has still cutted the diff size in half. Habile !
 
(*) https://github.com/OCA/account-fiscal-rule/pull/74/files#diff-8a5abf7050fe0075d6f0dd15f92db8a6

**Comment by bealdav** [comment]
Ok, I've done that I could with that I've understood. I can't do more on this subject.

I accept your proposal to finish/retry/fix that on Akretion repo.

Thanks for your time

**Comment by legalsylvain** [comment]
OK. I'll finish the changes in the next days. Thanks for your work.
regards.

**Comment by bealdav** [comment]
Hi @legalsylvain any news on this side ?

Maybe not time for you for this task ? Maybe we can merge as it ?

Thanks

**Review by legalsylvain — APPROVED** [review]
done. can be merged, once all tests have passed.

**Comment by bealdav** [comment]
Ahah. Tu es dur en affaire !

**Comment by rvalyi** [comment]
thanks for the cleanup guys, still ok for me

**Comment by bealdav** [comment]
Merci beaucoup

**Pull request merged**